Ira/my-app-video/src/redux/movies/movieSaga.ts

```diff
@@ -0,0 +1,78 @@
+import {takeLatest, call, put, all} from 'redux-saga/effects';
+import {MovieActionTypes} from "./actionTypes";
+import {baseUrl} from "../../constants/baseUrl";
+import {
+  fetchMoviesSuccess,
+  fetchMoviesError,
+  IFetchMovies,
+  IFetchMovieById,
+  fetchMovieByIdSuccess,
+} from "./actions";
+
+function* fetchMoviesSaga(action: IFetchMovies) {
+  const limit = 9;
+  const page = (action.payload?.page || 1) - 1;
+  const offset: number = page * limit + 1;
+  let url = `${baseUrl}/movies?limit=${limit}&offset=${offset}&sortOrder=desc`;
+
+  if (action.payload === undefined) {
+    url += `&sortBy=release_date`;
+  }
+
+  if (action.payload?.search) {
+    url += `&search=${action.payload?.search}`;
+  }
+
+  if (action.payload?.searchBy) {
+    url += `&searchBy=${action.payload?.searchBy}`;
+  }
+
+  if (action.payload?.sortBy) {
+    url += `&sortBy=${action.payload?.sortBy}`;
+  }
+  
+  if (action.payload?.searchBy === 'title') {
+    url = `${baseUrl}/movies?limit=${limit}&search=${action.payload?.search}&searchBy=${action.payload?.searchBy}`;
+  }
+
+  try {
+    const movies = yield call(() => fetch(`${url}`).then(res => res.json()));
+
+    yield put(fetchMoviesSuccess({films: movies.data, total: movies.total}));
+  } catch (e) {
+    yield put(fetchMoviesError());
+  }
+}
+
+const fetchMoviesSub = () => {
+  return takeLatest(MovieActionTypes.FETCH_MOVIES, fetchMoviesSaga);
+}
+
+function* fetchMovieByIdSaga(action: IFetchMovieById) {
+
+  try {
+    const movie = yield call(() => fetch(`${baseUrl}/movies/${action.payload}`).then(res => res.json()));
+
+    yield put(fetchMovieByIdSuccess(movie));
+
+    const search = movie.genres[0];
+    const url = `${baseUrl}/movies?limit=9&search=${search}&searchBy=genres&sortBy=release_date`;
+    const sortedMovies = yield call(() => fetch(`${url}`).then(res => res.json()));
+
+    yield put(fetchMoviesSuccess({films: sortedMovies.data, total: sortedMovies.total}));
```

Comment:
don't like that you do it in fetchMovieByIdSaga saga. Really confusing. I suppose that I fetch single movie, but instead I also fetch some list of movies. And how you handle pagination here ? I don't see how you fetch **next** movies by **genre** ?